### DISCUSSION TITLE
startsharp background jobs

### DISCUSSION THREAD

**Discussion by AviBit** [serenity-is/Serenity#6070] [Help]
Hello 
I wanted to buy the startsharp version, but before I do that wanted to know if background task to generate pdf reports feature is fully working

I need a 
Background task system 
generate pdf report
then send as email attachment

**Comment by VictorTomaili** [comment]
please check our website https://serenity.is/

**Comment by AviBit** [comment]
yes i did, just wanted to know if this sample implemented in the paid version, ill buy it if its there 

https://serenity.is/Docs/startsharp/features/background-pdf-reports

**Reply by VictorTomaili** [reply]
So, send an email to us with any questions related to the StartSharp

**Comment by kilroyFR** [comment]
I have a question related to these background jobs (BackgroundTaskManager). 
I have scheduled a dailybackground job (host is IIS) that collect some statistical indicators every day (so update date of last generation and generation a new set every day). I did not use GUI application for some days and noticed the indicators were not calculated. Looks like 'background tasks" was paused during that time. I though they behave like worker thread, working as long as application pool was running.

Anyone else did encounter this behavior ?

**Reply by VictorTomaili** [reply]
If the IIS put your website sleep, yes, the background job will be stopped.

**Reply by kilroyFR** [reply]
True thanks Victor. Just checked my Idle Timeout value was set to 20 (not the recommended 0 value to disable idle timeout). Many thanks
